src/fheroes2/maps/maps_tiles.cpp

```diff
@@ -1917,42 +1898,47 @@ void Maps::Tiles::swap( TilesAddon & addon ) noexcept
-    std::swap( addon._hasObjectAnimation, _hasObjectAnimation );
-    std::swap( addon._isMarkedAsRoad, _isMarkedAsRoad );
 }
 
 StreamBase & Maps::operator<<( StreamBase & msg, const TilesAddon & ta )
 {
     using ObjectIcnTypeUnderlyingType = std::underlying_type_t<decltype( ta._objectIcnType )>;
 
-    return msg << ta._layerType << ta._uid << static_cast<ObjectIcnTypeUnderlyingType>( ta._objectIcnType ) << ta._hasObjectAnimation << ta._isMarkedAsRoad
-               << ta._imageIndex;
+    return msg << ta._layerType << ta._uid << static_cast<ObjectIcnTypeUnderlyingType>( ta._objectIcnType ) << ta._imageIndex;
 }
 
 StreamBase & Maps::operator>>( StreamBase & msg, TilesAddon & ta )
 {
-    msg >> ta._layerType >> ta._uid;
+    msg >> ta._layerType;
+
+    static_assert( LAST_SUPPORTED_FORMAT_VERSION < FORMAT_VERSION_1009_RELEASE, "Remove the logic below." );
+    if ( Game::GetVersionOfCurrentSaveFile() < FORMAT_VERSION_1009_RELEASE ) {
+        ta._layerType = ( ta._layerType & 0x03 );
+    }
+
+    msg >> ta._uid;
 
     using ObjectIcnTypeUnderlyingType = std::underlying_type_t<decltype( ta._objectIcnType )>;
     static_assert( std::is_same_v<ObjectIcnTypeUnderlyingType, uint8_t>, "Type of _objectIcnType has been changed, check the logic below" );
 
-    static_assert( LAST_SUPPORTED_FORMAT_VERSION < FORMAT_VERSION_1001_RELEASE, "Remove the logic below." );
-    if ( Game::GetVersionOfCurrentSaveFile() < FORMAT_VERSION_1001_RELEASE ) {
-        ObjectIcnTypeUnderlyingType objectIcnType = MP2::OBJ_ICN_TYPE_UNKNOWN;
-        msg >> objectIcnType;
+    static_assert( LAST_SUPPORTED_FORMAT_VERSION < FORMAT_VERSION_1009_RELEASE, "Remove the logic below." );
+    if ( Game::GetVersionOfCurrentSaveFile() < FORMAT_VERSION_1009_RELEASE ) {
+        static_assert( LAST_SUPPORTED_FORMAT_VERSION < FORMAT_VERSION_1001_RELEASE, "Remove the logic below." );
+        if ( Game::GetVersionOfCurrentSaveFile() < FORMAT_VERSION_1001_RELEASE ) {
+            ObjectIcnTypeUnderlyingType objectIcnType = MP2::OBJ_ICN_TYPE_UNKNOWN;
+            msg >> objectIcnType;
 
-        ta._objectIcnType = static_cast<MP2::ObjectIcnType>( objectIcnType >> 2 );
-
-        ta._hasObjectAnimation = ( objectIcnType & 1 ) != 0;
-        ta._isMarkedAsRoad = ( objectIcnType & 2 ) != 0;
-    }
-    else {
-        ObjectIcnTypeUnderlyingType objectIcnType = MP2::OBJ_ICN_TYPE_UNKNOWN;
-        msg >> objectIcnType;
+            ta._objectIcnType = static_cast<MP2::ObjectIcnType>( objectIcnType >> 2 );
+        }
+        else {
+            ObjectIcnTypeUnderlyingType objectIcnType = MP2::OBJ_ICN_TYPE_UNKNOWN;
+            msg >> objectIcnType;
 
-        ta._objectIcnType = static_cast<MP2::ObjectIcnType>( objectIcnType );
+            ta._objectIcnType = static_cast<MP2::ObjectIcnType>( objectIcnType );
 
-        msg >> ta._hasObjectAnimation >> ta._isMarkedAsRoad;
+            bool temp;
+            msg >> temp >> temp;
+        }
     }
 
```

Comment:
In the case if we load the `FORMAT_VERSION_1009_RELEASE` save version we should not skip loading the `_objectIcnType`:
```suggestion
    else {
        ObjectIcnTypeUnderlyingType objectIcnType = MP2::OBJ_ICN_TYPE_UNKNOWN;
        msg >> objectIcnType;

        ta._objectIcnType = static_cast<MP2::ObjectIcnType>( objectIcnType );
    }

```